rcl/test/rcl/test_graph.cpp

```diff
@@ -329,13 +341,304 @@ check_graph_state(
+
+    ret = rcl_node_init(remote_node_ptr, remote_node_name, "", this->remote_context_ptr,
+        &node_options);
+    ASSERT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    sub_func = std::bind(rcl_get_subscriber_names_and_types_by_node,
+        std::placeholders::_1,
+        &this->allocator,
+        false,
+        std::placeholders::_2,
+        "/",
+        std::placeholders::_3);
+    pub_func = std::bind(rcl_get_publisher_names_and_types_by_node,
+        std::placeholders::_1,
+        &this->allocator,
+        false,
+        std::placeholders::_2,
+        "/",
+        std::placeholders::_3);
+    service_func = std::bind(rcl_get_service_names_and_types_by_node,
+        std::placeholders::_1,
+        &this->allocator,
+        std::placeholders::_2,
+        "/",
+        std::placeholders::_3);
+    WaitForAllNodesAlive();
+  }
+
+  void TearDown() override
+  {
+    rcl_ret_t ret;
+    CLASSNAME(TestGraphFixture, RMW_IMPLEMENTATION) ::TearDown();
+    ret = rcl_node_fini(this->remote_node_ptr);
+
+    delete this->remote_node_ptr;
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    RCUTILS_LOG_DEBUG_NAMED(ROS_PACKAGE_NAME, "Tearing down class");
+
+    ret = rcl_shutdown(this->remote_context_ptr);
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    ret = rcl_context_fini(this->remote_context_ptr);
+    EXPECT_EQ(RCL_RET_OK, ret) << rcl_get_error_string().str;
+    delete this->remote_context_ptr;
+  }
+
+  void WaitForAllNodesAlive()
+  {
+    rcl_ret_t ret;
+    rcutils_string_array_t node_names = rcutils_get_zero_initialized_string_array();
+    rcutils_string_array_t node_namespaces = rcutils_get_zero_initialized_string_array();
+    OSRF_TESTING_TOOLS_CPP_SCOPE_EXIT({
+      ret = rcutils_string_array_fini(&node_names);
+      ASSERT_EQ(RCUTILS_RET_OK, ret);
+      ret = rcutils_string_array_fini(&node_namespaces);
+      ASSERT_EQ(RCUTILS_RET_OK, ret);
+    });
+    // wait for all 3 nodes to be discovered: remote_node, old_node, node
+    size_t attempts = 0;
+    size_t max_attempts = 4;
+    while (node_names.size < 3) {
+      std::this_thread::sleep_for(std::chrono::seconds(1));
+      ret = rcl_get_node_names(this->remote_node_ptr, allocator, &node_names, &node_namespaces);
+      attempts++;
+      ASSERT_LE(attempts, max_attempts) << "Unable to attain all required nodes";
+    }
+  }
+
+  /**
+   * Verify the number of subsystems each node should have.
+   *
+   * @param node_state expected state of node
+   * @param remote_node_state expected state of remote node
+   */
+  void VerifySubsystemCount(
+    const expected_node_state && node_state,
+    const expected_node_state && remote_node_state) const
+  {
+    std::vector<rcl_node_t *> node_vec;
+    node_vec.push_back(this->node_ptr);
+    if (!(is_opensplice || is_connext)) {
+      // @todo: opensplice and connext cannot discover data about the current node
```

Comment:
We use `TODO(name)` for our todo comments.  Also, can you maybe elaborate a little more on what this means?  Does that mean the graph stuff isn't going to work on opensplice or connext at all?

Reply:
This means that when inspecting the node from the perspective of the current node, for the current node, nothing is returned for opensplice/rti_connext. This is due to the unique data structures each DDS implementation has implemented. It does not affect the node graph API, node graph CLI, or security generation.
This will be added as a lower priority change after release.